component/publisher-client/src/main/java/org/wso2/am/analytics/publisher/reporter/cloud/DefaultResponseMetricEventBuilder.java

```diff
@@ -64,7 +75,31 @@ public boolean validate() throws MetricReportingException {
 
     @Override
     protected Map<String, Object> buildEvent() {
-        eventMap.put(Constants.EVENT_TYPE, "response");
+        eventMap.put(Constants.EVENT_TYPE, Constants.RESPONSE_EVENT_TYPE);
+        // userAgent raw string is not required and removing
+        String userAgentHeader = (String) eventMap.remove(Constants.USER_AGENT_HEADER);
+        // userAgentHeader will not null since it is already validated
+        setUserAgentProperties(userAgentHeader);
         return eventMap;
     }
+
+    private void setUserAgentProperties(String userAgentHeader) {
+        if (uaParser == null) {
```

Comment:
@fazlan-nazeem Can stream layer handle if we send an event without os and browser properties?

Reply:
we don't need that. But anyway I added unknown value for them,

Reply:
@tishan89 so far we haven't had issues related to empty values at the streaming layer. But I will test this further and see.